internal/datasources/app_endpoints.go

```diff
@@ -0,0 +1,127 @@
+package datasources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/app_endpoints"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// NewAppEndpoint is used in (p *capellaProvider) DataSources for building the provider.
+func NewAppEndpoint() datasource.DataSource {
+	return &AppEndpoint{}
+}
+
+// Metadata returns the App Endpoints data source type name.
+func (a *AppEndpoint) Metadata(
+	_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_app_endpoints"
+}
+
+// Configure defines the schema for the App Endpoints data source.
+func (a *AppEndpoint) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	a.Data = data
+}
+
+// Read refreshes the Terraform state with the latest App Endpoints configs.
```

Comment:
it is not refreshing state.  that happens with read handler in resource.  

this read is just a list